frontend/widgets/OBSBasic_SceneItems.cpp

```diff
@@ -369,7 +369,21 @@ void OBSBasic::SetDeinterlacingMode()
 	OBSSceneItem sceneItem = GetCurrentSceneItem();
 	obs_source_t *source = obs_sceneitem_get_source(sceneItem);
 
+	obs_deinterlace_mode oldMode = obs_source_get_deinterlace_mode(source);
 	obs_source_set_deinterlace_mode(source, mode);
+
+	auto undo_redo = [](const std::string &uuid, obs_deinterlace_mode val) {
+		OBSSourceAutoRelease source = obs_get_source_by_uuid(uuid.c_str());
+		if (source)
+			obs_source_set_deinterlace_mode(source, val);
+	};
+
+	const char *uuid = obs_source_get_uuid(source);
+
+	QString str = QTStr("Undo.DeinterlacingMode");
+
+	undo_s.add_action(str.arg(obs_source_get_name(source)), std::bind(undo_redo, std::placeholders::_1, oldMode),
+			  std::bind(undo_redo, std::placeholders::_1, mode), uuid, uuid);
```

Comment:
```suggestion

    auto undoFunction = std::bind(undo_redo, std::placeholders::_1, oldMode);
    auto redoFunction = std::bind(undo_redo, std::placeholders::_1, mode);

	undo_s.add_action(str.arg(obs_source_get_name(source)), undoFunction, redoFunction, uuid, uuid);
```

I'd prefer if this were split up to make the code more readable (compilers will potentially optimise this in better ways anyway).

It's a bit unfortunate that the `uuid` argument is not bound directly as well, because it should not change between here and when the lambdas are invoked, but the system as-designed still requires a single string as "data" to be provided.